### PR TITLE
Added Slayer Command

### PR DESCRIPTION
**Pull request by HyxonQz**
Has overview aswell as specific slayer stats. Probably needs some backend polishing

### CODE REVIEW

File: src/instance/minecraft/commands/SlayerCommand.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import * as assert from 'node:assert'
+import { Slayer } from 'hypixel-api-reborn'
+import { ChatCommandContext, ChatCommandHandler } from '../common/ChatInterface'
+
+const Slayers: Record<string, string[]> = {
+  zombie: ['revenant', 'rev', 'zombie'],
+  spider: ['tarantula', 'tara', 'spider'],
```
Comment (aidn3) [inline src/instance/minecraft/commands/SlayerCommand.ts:7]: People call it `tar` as well.

File: src/instance/minecraft/commands/SlayerCommand.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  wolf: ['sven', 'wolf'],
+  enderman: ['voidgloom', 'eman', 'enderman'],
+  blaze: ['inferno', 'demonlord', 'blaze'],
+  vampire: ['riftstalker', 'bloodfiend', 'vamp', 'vampire'],
+  overview: ['all', 'summary']
+}
+
+export default {
+  name: 'Secrets',
```
Comment (aidn3) [inline src/instance/minecraft/commands/SlayerCommand.ts:16]: You probably means `Slayers`
```suggestion
  name: 'Slayers',
```

File: src/instance/minecraft/commands/SlayerCommand.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  triggers: ['slayer', 'sl', 'slyr'],
+  description: "Returns a player's slayer levels",
```
Comment (aidn3) [inline src/instance/minecraft/commands/SlayerCommand.ts:18]: Spell check: I think `player's slayer level` or `player's slayers levels`

File: src/instance/minecraft/commands/SlayerCommand.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      })
+
+    if (uuid == undefined) {
+      return `${context.username}, Invalid username! (given: ${givenUsername})`
+    }
+
+    const hypixelProfile = await context.clientInstance.app.hypixelApi.getPlayer(uuid)
+    assert(hypixelProfile)
+
+    const slayers = await context.clientInstance.app.hypixelApi
```
Comment (aidn3) [inline src/instance/minecraft/commands/SlayerCommand.ts:39]: Change it to `slayersProfile`. `Slayers` already used. Taking advantage of case-sensitive variables might be confusing to readers.

File: src/instance/minecraft/commands/SlayerCommand.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      .then((profiles) => profiles.find((p) => p.selected))
+      .then((response) => response?.members[uuid].slayer_bosses)
+    assert(slayers)
+
+    let chosenSlayer: string | undefined
+    for (const [key, names] of Object.entries(Slayers)) {
+      if (names.includes(givenSlayer.toLowerCase())) {
+        chosenSlayer = key
+      }
+    }
```
Comment (aidn3) [inline src/instance/minecraft/commands/SlayerCommand.ts:51]: Do the checking before you doing the mojang and hypixel api requests.

**Review by aidn3 — CHANGES_REQUESTED** [review]

File: src/instance/minecraft/commands/SlayerCommand.ts (hunk 3 above)
Comment (HyxonQz) [reply src/instance/minecraft/commands/SlayerCommand.ts:18]: done

File: src/instance/minecraft/commands/SlayerCommand.ts (hunk 4 above)
Comment (HyxonQz) [reply src/instance/minecraft/commands/SlayerCommand.ts:39]: done

File: src/instance/minecraft/commands/SlayerCommand.ts (hunk 1 above)
Comment (HyxonQz) [reply src/instance/minecraft/commands/SlayerCommand.ts:7]: done

File: src/instance/minecraft/commands/SlayerCommand.ts (hunk 5 above)
Comment (HyxonQz) [reply src/instance/minecraft/commands/SlayerCommand.ts:51]: Is this required? The command will default to 'overview' if the given slayer is not valid

**Review by aidn3 — APPROVED** [review]

**Pull request merged**